deploy/helm/nfd-operator/templates/deployment.yaml

```
@@ -4,78 +4,54 @@ metadata:
   labels:
     control-plane: controller-manager
     {{- include "node-feature-discovery-operator.labels" . | nindent 4 }}
-  name: {{ include "node-feature-discovery-operator.fullname" . }}-controller-manager
+  name: {{ include "node-feature-discovery-operator.name" . }}-controller-manager
```

Why change this fullname -> name?
```suggestion
  name: {{ include "node-feature-discovery-operator.fullname" . }}-controller-manager
```